### PR TITLE
Colorful

### PR DESCRIPTION
**Pull request by aminya**
Adds options for button color and background-color

Derived from https://github.com/suda/tool-bar/pull/21

### CODE REVIEW

**Comment by aminya** [comment]
Could you review this? @suda

**Comment by suda** [comment]
Hi @aminya I'm currently away for two weeks and unable to test it. Is it ok if I to it after I'm back?

**Comment by aminya** [comment]
> Hi @aminya I'm currently away for two weeks and unable to test it. Is it ok if I to it after I'm back?

Yes sure! Thanks

File: README.md [hunk 1]
```diff
@@ -206,12 +206,21 @@ export function consumeToolBar(getToolBar) {
     this.toolBar = null;
     // Teardown any stateful code that depends on tool bar ...
   });
+
+  // Colourful button and background
+  toolBar.addButton({
```
Comment (ericcornelissen) [inline README.md]: I would move this in front of `toolBar.onDidDestroy` for consistency

File: README.md [hunk 2]
```diff
@@ -206,12 +206,21 @@ export function consumeToolBar(getToolBar) {
     this.toolBar = null;
     // Teardown any stateful code that depends on tool bar ...
   });
+
+  // Colourful button and background
```
Comment (ericcornelissen) [inline README.md]: Consider changing the comment to be consistent with the other examples:

```suggestion
  // Text buttons can be colored
```

File: README.md [hunk 3]
```diff
@@ -206,12 +206,21 @@ export function consumeToolBar(getToolBar) {
...
+  toolBar.addButton({
+    icon: 'octoface',
+    callback: 'application:about',
+    tooltip: 'About Atom'
```
Comment (ericcornelissen) [inline README.md]: Missed a comma here

```suggestion
    tooltip: 'About Atom',
```

File: README.md [hunk 4]
```diff
@@ -206,12 +206,21 @@ export function consumeToolBar(getToolBar) {
...
+    color: 'red' // colors of the button
```
Comment (ericcornelissen) [inline README.md]: Consider changing the comment for clarity:

```suggestion
    color: 'red' // color of the text or icon
```

File: README.md [hunk 5]
```diff
@@ -226,6 +235,8 @@ The remaining properties
 `iconset` (defaults to `Octicons`),
 `data`, and
 `priority` (defaults `50`)
+`color`
+`background`
```
Comment (ericcornelissen) [inline README.md]: I think this should be _"`data`, `priority` (defaults `50`), `color`, and `background`"_ could you update it accordingly?

File: lib/tool-bar-button-view.js [hunk 6]
```diff
@@ -11,6 +11,13 @@ module.exports = class ToolBarButtonView {
     this.group = group;
     this.enabled = true;
 
+    if (options.color) {
```
Comment (ericcornelissen) [inline lib/tool-bar-button-view.js]: Minor thing & personal preference, but for maintainability I would suggest moving this to later in the function. E.g. after the [`options.text` if-statement](https://github.com/aminya/tool-bar/blob/09c85114a26cbcb7af78eb53a35253aa35e25139/lib/tool-bar-button-view.js#L62)? What do you think?

**Comment by aminya** [comment]
> 
> 
> Implementation looks good, I have some comments (mainly about the documentation) to improve the Pull Request 👍
> 
> Ps. you may consider adding your contributions to [the CHANGELOG](https://github.com/suda/tool-bar/blob/master/CHANGELOG.md)

All are fixed in https://github.com/suda/tool-bar/pull/278/commits/1f0e0bbc445a1f00fd0525ffcaa1400514dca5d6. Thanks for the comments.

**Comment by suda** [comment]
Thanks for the review Eric! Merging in :)

**Pull request merged**